### PR TITLE
Add codegen scripts and make required api changes

### PR DESCRIPTION
**Pull request by harryge00**
## Why are these changes needed?
To support clientset for RayCluster objects, we may use `code-gen` to generate clientset. With PR, you can run `hack/update-codegen.sh` to generate clientset. The generated code will be at `pkg/client/clientset` 

## Related issue number
#29 

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Comment by harryge00** [comment]
@Jeffwan

File: ray-operator/go.mod [hunk 1]
```diff
@@ -3,14 +3,38 @@ module github.com/ray-project/kuberay/ray-operator
 go 1.15
 
 require (
-	github.com/go-logr/logr v0.3.0
+	github.com/go-logr/logr v0.4.0
```
Comment (Jeffwan) [inline ray-operator/go.mod:6]: We don't necessarily need to change upgrade go mod.

**Comment by Jeffwan** [comment]
@harryge00 Do you mind giving me the collaborator permission to your repo and I can co-author with you in this PR? 
Or I can cherry-pick your changes to separate PR. 

Which one do you prefer?

**Comment by harryge00** [comment]
@Jeffwan I've invited you to my repo.

**Comment by Jeffwan** [comment]
@harryge00 I update the PR and manually generate clients but have not checked in. Please have a review

File: ray-operator/go.sum [hunk 2]
```diff
@@ -650,6 +650,8 @@ k8s.io/client-go v0.19.2/go.mod h1:S5wPhCqyDNAlzM9CnEdgTGV4OqhsW3jGO1UM1epwfJA=
 k8s.io/client-go v0.19.14 h1:D3V4mC6lQN3ZaABhOznxZXSsxMR5cdChjeMP4/pY4H8=
 k8s.io/client-go v0.19.14/go.mod h1:iWwXK/etTTPyssCycvgdQ9d6i0k93HXLtksUQ6A67WE=
 k8s.io/code-generator v0.19.2/go.mod h1:moqLn7w0t9cMs4+5CQyxnfA/HV8MF6aAVENF+WZZhgk=
+k8s.io/code-generator v0.19.14 h1:EFU42Hj0+FBZciWedyRtFHsPu75IaWca0slGB1LlBqk=
```
Comment (harryge00) [inline ray-operator/go.sum:653]: Shall we unify the `code-generator` version?

Comment (Jeffwan) [reply ray-operator/go.sum:653]: @harryge00 

k8s.io/code-generator is pinned to v0.19.14. 0.19.2 is from some other indirect dependencies I think. That's impossible to unify all version in go.sum. Let's try to make sure we use same minor version in go.mod

image

**Pull request merged**